tycho-indexer/src/extractor/dynamic_contract_indexer/component_metadata.rs

```diff
@@ -269,6 +311,31 @@ pub trait MetadataResponseParser: Send + Sync {
     ) -> Result<MetadataValue, MetadataError>;
 }
 
+pub struct MetadataResponseParserRegistry {
+    // Map of protocol name to responseparser
+    parsers: HashMap<String, Box<dyn MetadataResponseParser>>,
+}
+
+impl MetadataResponseParserRegistry {
+    pub fn new() -> Self {
+        Self { parsers: HashMap::new() }
+    }
+
+    pub fn register_parser(
+        &mut self,
+        protocol_name: String,
```

Comment:
Shouldn't the parser be linked 1-1 to a request generator / provider and not to the protocol?